### PR TITLE
1.0.2 - Background Syncing for Skyn Revision

### PR DESCRIPTION
**Pull request by thertzelle**
### What

This changing a few of the methods from the previous PR. New things learned and updated the way we listen and react to events.

### Why

So we can do device-based syncing. (The device tells us when to sync).

### CODE REVIEW

File: BACDeviceMangementProtocol.h [hunk 1]
```diff
@@ -33,6 +33,7 @@
 -(void)startScan;
 // Scan for Skyn BacTrack devices
 -(void)scanForSkyn;
+-(void)searchForSkyn;
```
Comment (thertzelle) [inline BACDeviceMangementProtocol.h:36]: New method for Searching for the specific skyn tx that is advertising mode.

File: BacTrackAPI.m [hunk 2]
```diff
@@ -217,6 +217,11 @@ -(void)scanForSkyn
     [cmanager scanForPeripheralsWithServices:scanUdids options:0];
 }
 
+-(void)searchForSkyn
+{
+    [cmanager scanForPeripheralsWithServices:@[[CBUUID UUIDWithString:SKYN_SERIAL_GATT_TX_CHAR_UUID]] options:0];
```
Comment (thertzelle) [inline BacTrackAPI.m:222]: All we are going to do is queue it within the bluetooth manager (cmanager) to look out for some device that advertises with this UUID

File: BacTrackAPI.m [hunk 3]
```diff
@@ -631,6 +636,7 @@ -(void)connectTimeout
 /****************************************************************************/
 
 - (void)centralManager:(CBCentralManager *)central willRestoreState:(NSDictionary<NSString *,id> *)dict {
+    // Not required to be implemented for our use cases. But required by the framework when using background modes.
```
Comment (thertzelle) [inline BacTrackAPI.m:639]: Added an additional comment per the last PR, just to clearly document why this is here and why its un-implemented today.

File: BacTrackAPI.m [hunk 4]
```diff
@@ -792,7 +798,14 @@ - (void)centralManager:(CBCentralManager *)central didDiscoverPeripheral:(CBPeri
     breathalyzer.uuid = peripheral.identifier.UUIDString;
     
     [self getDiscoveredBreathalyzerType:breathalyzer withServices:[advertisementData objectForKey:@"kCBAdvDataServiceUUIDs"]];
-    
+
+    // Listen for Skyn Specific Notifications
+    if ([[advertisementData objectForKey:@"kCBAdvDataServiceUUIDs"] containsObject: [CBUUID UUIDWithString:SKYN_SERIAL_GATT_TX_CHAR_UUID]])
```
Comment (thertzelle) [inline BacTrackAPI.m:803]: If we found a device with this UUID in it, we should let the client know!

File: Skyn/BacTrackAPI_Skyn.m [hunk 5]
```diff
@@ -365,11 +365,6 @@ - (void)peripheral:(CBPeripheral *)peripheral didUpdateValueForCharacteristic:(C
         if ([self.delegate respondsToSelector:@selector(BacTrackSerial:)])
             [self.delegate BacTrackSerial:serialNumber];
     }
-    else if (characteristic == mCharacteristicSerialTx)
-    {
-        if ([self.delegate respondsToSelector:@selector(BacTrackSkynSyncRequest)])
-            [self.delegate BacTrackSkynSyncRequest];
-    }
```
Comment (thertzelle) [inline Skyn/BacTrackAPI_Skyn.m:372]: No longer needed at a characteristic level.

File: Skyn/BacTrackAPI_Skyn.m [hunk 6]
```diff
@@ -570,7 +565,6 @@ - (void)peripheral:(CBPeripheral *)peripheral didDiscoverCharacteristicsForServi
                 mCharacteristicSerialTx = characteristic;
         }
         [mPeripheral setNotifyValue:YES forCharacteristic:mCharacteristicSerialRx];
-        [mPeripheral setNotifyValue:YES forCharacteristic:mCharacteristicSerialTx];
```
Comment (thertzelle) [inline Skyn/BacTrackAPI_Skyn.m:573]: No longer needed at a characteristic level.

**Review by jlstr — APPROVED** [review]

**Review by samcdavid — APPROVED** [review]

**Pull request merged**